core/aws_ddk_core/stages/glue_transform.py

```diff
@@ -61,20 +68,54 @@ def __init__(
             Identifier of the stage
         environment_id : str
             Identifier of the environment
-        job_name : str
-            Name of the Glue job to run
-        crawler_name : str
-            Name of the Glue crawler to run
+        executable : Optional[JobExecutable]
+            The job executable properties
+        job_role : Optional[IRole]
+            The job execution role
+        job_name : Optional[str]
+            The name of the Glue job to run
+        database_name : Optional[str]
+            The name of the database in which the crawler's output is stored
+        crawler_role : Optional[IRole]
+            The crawler execution role
+        targets : Optional[TargetsProperty]
+            A collection of targets to crawl
+        crawler_name : Optional[str]
+            The name of the Glue crawler to run
         job_args : Optional[Dict[str, Any]]
-            Glue job arguments
+            The input arguments to the Glue job
         state_machine_input : Optional[Dict[str, Any]]
-            Input of the state machine
+            The input dict to the state machine
         """
         super().__init__(scope, id)
 
         self._state_machine_input: Optional[Dict[str, Any]] = state_machine_input
         self._event_detail_type: str = f"{id}-event-type"
 
+        # If Glue job name is not supplied, create one
+        self._job: Optional[IJob] = None
+        if not job_name:
+            self._job = GlueFactory.job(
+                self,
+                id=f"{id}-job",
+                environment_id=environment_id,
+                executable=executable,
+                role=job_role,
+            )
+            job_name = self._job.job_name
+
+        # If Glue crawler name is not supplied, create one
+        self._crawler: Optional[CfnCrawler] = None
```

Comment:
Are we keeping crawlers out of resource factory for a specific reason ? I know that they can be configured quite differently depending on use so maybe it's hard to have a relatively generic construct for it ?

Reply:
I guess being able to configure crawler paths depending on env would be a good idea